### PR TITLE
[Admin] Fix post-details, add feature sort posts, remove feature search by content

### PR DESCRIPTION
**Pull request by ghost**
## Description
Remodel Post Details Page and Add Feature Sort Posts By Username or Product Name
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`

1. explain select count(*) as aggregate from `posts` inner join `users` on `posts`.`user_id` = `users`.`id` inner join `products` on `posts`.`product_id` = `products`.`id` where `posts`.`deleted_at` is null\G

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: posts
>    partitions: NULL
>          type: ALL
> possible_keys: posts_product_id_foreign,posts_user_id_foreign
>           key: NULL
>       key_len: NULL
>           ref: NULL
>          rows: 80
>      filtered: 10.00
>         Extra: Using where
> *************************** 2. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: users
>    partitions: NULL
>          type: eq_ref
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: product_tiki.posts.user_id
>          rows: 1
>      filtered: 100.00
>         Extra: Using index
> *************************** 3. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: products
>    partitions: NULL
>          type: eq_ref
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: product_tiki.posts.product_id
>          rows: 1
>      filtered: 100.00
>         Extra: Using index
> 3 rows in set, 1 warning (0.00 sec)

2. explain select * from `posts` inner join `users` on `posts`.`user_id` = `users`.`id` inner join `products` on `posts`.`product_id` = `products`.`id` where `posts`.`deleted_at` is null order by `username` desc limit 5 offset 0\G

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: users
>    partitions: NULL
>          type: ALL
> possible_keys: PRIMARY
>           key: NULL
>       key_len: NULL
>           ref: NULL
>          rows: 10
>      filtered: 100.00
>         Extra: Using temporary; Using filesort
> *************************** 2. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: posts
>    partitions: NULL
>          type: ALL
> possible_keys: posts_product_id_foreign,posts_user_id_foreign
>           key: NULL
>       key_len: NULL
>           ref: NULL
>          rows: 80
>      filtered: 1.25
>         Extra: Using where; Using join buffer (Block Nested Loop)
> *************************** 3. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: products
>    partitions: NULL
>          type: eq_ref
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: product_tiki.posts.product_id
>          rows: 1
>      filtered: 100.00
>         Extra: NULL
> 3 rows in set, 1 warning (0.00 sec)

3. explain select * from `users` where `users`.`id` in ('7')\G

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: users
>    partitions: NULL
>          type: const
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: const
>          rows: 1
>      filtered: 100.00
>         Extra: NULL
> 1 row in set, 1 warning (0.00 sec)

4. explain select * from `user_info` where `user_info`.`user_id` in ('7')\G

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: user_info
>    partitions: NULL
>          type: ref
> possible_keys: user_info_user_id_foreign
>           key: user_info_user_id_foreign
>       key_len: 4
>           ref: const
>          rows: 1
>      filtered: 100.00
>         Extra: NULL
> 1 row in set, 1 warning (0.00 sec)

5. explain select * from `products` where `products`.`id` in ('2', '9', '14', '17') and `products`.`deleted_at` is null\G

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: products
>    partitions: NULL
>          type: range
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: NULL
>          rows: 4
>      filtered: 10.00
>         Extra: Using where
> 1 row in set, 1 warning (0.00 sec)

### CODE REVIEW

File: app/Http/Controllers/Admin/PostController.php [hunk 1]
```diff
@@ -20,14 +20,28 @@ class PostController extends Controller
...
+        $posts = Post::when(isset($request->post_status), function ($query) use ($request) {
             return $query->where('status', '=', $request->post_status);
         })
-        ->with(['user','product'])->paginate($perPage);
-        $posts->appends(request()->query());
-        $data['posts'] = $posts;
+        ->with(['user' => function ($query) {
+            return $query->with('userInfo');
+        },'product']);
+        \DB::enableQueryLog();
```
Comment (at-khadinh) [inline app/Http/Controllers/Admin/PostController.php]: bỏ đi Phương, với mấy cái cmt dd luôn

File: app/Http/Controllers/Admin/PostController.php [hunk 2]
```diff
@@ -20,14 +20,26 @@ class PostController extends Controller
...
-        })->when(isset($request->post_status), function ($query) use ($request) {
+        $posts = Post::when(isset($request->post_status), function ($query) use ($request) {
             return $query->where('status', '=', $request->post_status);
         })
-        ->with(['user','product'])->paginate($perPage);
-        $posts->appends(request()->query());
-        $data['posts'] = $posts;
+        ->with(['user' => function ($query) {
+            return $query->with('userInfo');
+        },'product']);
```
Comment (at-khadinh) [inline app/Http/Controllers/Admin/PostController.php]: convention Phương

File: app/Http/Controllers/Admin/PostController.php [hunk 3]
```diff
@@ -20,14 +20,26 @@ class PostController extends Controller
...
+        ->with(['user' => function ($query) {
+            return $query->with('userInfo');
+        }, 'product']);
+
+        if (isset($request->sortBy) && isset($request->dir)) {
+            $postSort = $posts->join('users', 'posts.user_id', 'users.id')
+                        ->join('products', 'posts.product_id', 'products.id')
+                        ->orderBy($request->sortBy, $request->dir)
+                        ->paginate($perPage);
+            $postSort->appends(request()->query());
```
Comment (at-huenguyen) [inline app/Http/Controllers/Admin/PostController.php]: để ```$postSort->appends(request()->query());
$data['posts'] = $postSort;```
ra ngoài `if..else`, dùng chung cho `if` vs `else` lun em.

File: resources/views/admin/pages/posts/index.blade.php [hunk 4]
```diff
@@ -48,15 +45,35 @@
...
+
             <div class="x_content" class="list-table">
               <table class="table table-hover">
                 <thead>
                   <tr>
-                    <th class="col-md-1">{{ __('post.admin.list.user_col') }}</th>
-                    <th class="col-md-2">{{ __('post.admin.list.product_col') }}</th>
+                    <th class="col-md-2">{{ __('post.admin.list.avatar_col') }}</th>
+                    <th class="col-md-2">{{ __('post.admin.list.user_col') }}
+                      @if (app('request')->input('dir') == 'DESC' && app('request')->input('sortBy') == 'username')
```
Comment (at-huenguyen) [inline resources/views/admin/pages/posts/index.blade.php]: ko hardcode nha em, khai báo `DESC`, `username` vs những liên quan khác ở chỗ khác rồi gọi lại nha

**Review by ghost — CHANGES_REQUESTED** [review]
check cmt nha em

File: resources/views/admin/pages/posts/index.blade.php (hunk 4 above)
Comment (at-phuongvu) [reply resources/views/admin/pages/posts/index.blade.php]: dạ cái này em commit r mà chưa push lên ạ :D

File: app/Http/Controllers/Admin/PostController.php [hunk 5]
```diff
@@ -20,14 +20,24 @@ class PostController extends Controller
...
         $perPage = config('define.post.limit_rows');
-        $posts = Post::when(isset($request->content), function ($query) use ($request) {
-            return $query->where('content', 'like', "%$request->content%");
-        })->when(isset($request->post_status), function ($query) use ($request) {
+        $posts = Post::when(isset($request->post_status), function ($query) use ($request) {
             return $query->where('status', '=', $request->post_status);
         })
-        ->with(['user','product'])->paginate($perPage);
+        ->with(['user' => function ($query) {
+            return $query->with('userInfo');
```
Comment (at-huenguyen) [inline app/Http/Controllers/Admin/PostController.php]: em muốn with thêm `userInfo` thì làm như vại lun:
```
Post::...
->with(['user.userInfo', 'product']);
```

File: app/Http/Controllers/Admin/PostController.php [hunk 6]
```diff
@@ -43,13 +53,18 @@ public function index(Request $request)
     public function show(Request $request, $id)
     {
         $perPage = config('define.post.limit_rows');
-        $comments = Post::find($id)->comments()->when(isset($request->content), function ($query) use ($request) {
+        $post = Post::with(['user' => function ($query) {
+            return $query->with('userInfo');
```
Comment (at-huenguyen) [inline app/Http/Controllers/Admin/PostController.php]: giống cmt ở dòng 27 nha

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**